src/main/java/uk/ac/ebi/atlas/search/GeneSearchDao.java

```diff
@@ -138,58 +143,53 @@ public GeneSearchDao(NamedParameterJdbcTemplate namedParameterJdbcTemplate,
         );
     }
 
-    // Returns inferred cell types and organism parts for each experiment accession
-    public Map<String, Map<String, List<String>>>
-    getFacets(List<String> cellIds, SingleCellAnalyticsSchemaField... singleCellAnalyticsSchemaFields) {
-        List<SingleCellAnalyticsSchemaField> subFacetFields = Arrays.asList(singleCellAnalyticsSchemaFields);
+    // Returns all the metadata values for each experiment accession, given a subset of metadata types
+    public Map<String, Map<String, List<String>>> getFacets(List<String> cellIds, String... metadataTypes) {
+        List<String> facetFieldValues = Arrays.asList(metadataTypes);
+
+        SolrJsonFacetBuilder<SingleCellAnalyticsCollectionProxy> characteristicValueFacet =
+                new SolrJsonFacetBuilder<SingleCellAnalyticsCollectionProxy>()
+                        .setFacetField(CHARACTERISTIC_VALUE)
+                        .isNestedFacet(true);
+
+        SolrJsonFacetBuilder<SingleCellAnalyticsCollectionProxy> characteristicNameFacet =
+                new SolrJsonFacetBuilder<SingleCellAnalyticsCollectionProxy>()
+                        .setFacetField(CHARACTERISTIC_NAME)
+                        .addSubFacets(ImmutableList.of(characteristicValueFacet))
+                        .isNestedFacet(true);
+
+        SolrJsonFacetBuilder<SingleCellAnalyticsCollectionProxy> facets =
+                new SolrJsonFacetBuilder<SingleCellAnalyticsCollectionProxy>()
+                        .setFacetField(EXPERIMENT_ACCESSION)
+                        .addSubFacets(ImmutableList.of(characteristicNameFacet));
```

Comment:
Funky! :rocket:

Reply:
Haha, in a good way? I think it's a little bit verbose and can be quite repetitive, which is why I made a lot of options default.

Reply:
I'm a convert! Helps a little with the verbosity.

![Screenshot 2019-05-13 at 15 23 38](https://user-images.githubusercontent.com/5923538/57629115-27ce8900-7593-11e9-8d69-d9c6de7660a6.png)

Reply:
Funky in a good way. Also, welcome to the dark side. :wink: